spec/01-unit/21-dns-client/02-client_spec.lua

```diff
@@ -36,27 +36,32 @@ end
 
 describe("[DNS client]", function()
 
-  local client, resolver, query_func
+  local client, resolver
 
   before_each(function()
     client = require("kong.resty.dns.client")
     resolver = require("resty.dns.resolver")
 
-    -- you can replace this `query_func` upvalue to spy on resolver query calls.
+    -- you can replace this `resolver.query_func`  to spy on resolver query calls.
```

Comment:
What does this comment mean? I don't understand.

Reply:
This means the function is spied(hooked) to inspect the internal behavior of the DNS resolver, and the hook function can be changed. I'm not the one who wrote this originally.